python/ruff/__main__.py

```diff
@@ -3,23 +3,29 @@
 import sysconfig
 from pathlib import Path
 
-RUFF_PATHS = [
-    Path(sysconfig.get_config_var("userbase")) / "bin" / "ruff",
-    Path(sysconfig.get_path("scripts")) / "ruff",
-]
-
 
 def find_ruff_bin() -> Path:
     """Return the ruff binary path."""
-    for ruff_path in RUFF_PATHS:
+    ruff_path = Path(sysconfig.get_path("scripts")) / "ruff"
+    if ruff_path.is_file():
+        return ruff_path
+
+    if sys.version_info >= (3, 10):
+        ruff_path = (
+            Path(
+                sysconfig.get_path(
+                    "scripts",
+                    scheme=sysconfig.get_preferred_scheme("user"),
+                ),
+            )
+            / "ruff"
```

Comment:
We can also try falling back to `Path(sysconfig.get_config_var("userbase")) / "bin" / "ruff"`, it shouldn't hurt, it just might be wrong on Windows.